### DISCUSSION TITLE
Solving Attribute cannot be applied to a parameter because it does not contain the 'Attribute::TARGET_PARAMETER' flag

### DISCUSSION THREAD

**Discussion by kmizzi** [spatie/laravel-data#629] [General]
Does anyone know how to solve this PHPStorm warning?:

Attribute cannot be applied to a parameter because it does not contain the 'Attribute::TARGET_PARAMETER' flag 

Example usage:

public function saveWithRelations(#[DataCollectionOf(ProductDto::class)] DataCollection $data): Collection

**Comment by czuniga9** [comment]
[DataCollectionOf](https://github.com/spatie/laravel-data/blob/main/src/Attributes/DataCollectionOf.php#L9) only has `Attribute::TARGET_PROPERTY` so it can be used to target class properties. So from what I understand, constructor parameters are also properties thanks to [property promotion](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion) but for regular functions the parameters can't be properties.

If you need this for autocompletion, you could use phpdoc:

```php
    /**
     * @param DataCollection<array-key, ProductDto> $data
     */
    public function saveWithRelations(DataCollection $data)
    {
    }
```

**Reply by kmizzi** [reply]
Mainly I don't want the IDE to complain, but I can try this!
